### PR TITLE
API to assign 'formsflow-client' group to BCSC and BCeID users

### PR DESCRIPTION
**Pull request by jaspalsingh-aot**
# Description

Please provide a summary of the change and the issue fixed. Please include relevant context. List dependency changes.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist

- [ ] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have already been accepted and merged

### CODE REVIEW

**Review by nikhila-aot — CHANGES_REQUESTED** [review]

File: backend/users/src/app/services/KeycloakService.service.ts [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+              Authorization: `Bearer ${accessToken}`,
+              'Content-Type': 'application/json',
+            },
+            params: {
+              search: groupName,
+            },
+          });
+    
+          // Find group ID by name
+          const group = this.findGroupIdByName(response.data, groupName);
```
[the remark this answers is not in the document]
Comment (jaspalsingh-aot) [reply backend/users/src/app/services/KeycloakService.service.ts:78]: implemented method is faster.

File: backend/users/src/app/controllers/users.controller.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import { Controller, Post, Body, HttpException, HttpStatus  } from '@nestjs/common';
+import { Unprotected } from 'nest-keycloak-connect';
+import { KeycloakService } from 'src/app/services/KeycloakService.service';
+import { AddUserToGroupDto } from 'src/app/dto/addUserToGroup';
+
+@Controller('users')
+@Unprotected()
+export class UsersController {
```
[the remark this answers is not in the document]
Comment (jaspalsingh-aot) [reply backend/users/src/app/controllers/users.controller.ts]: Changed to UserController and User.controller.ts

File: backend/users/src/app/services/KeycloakService.service.ts [hunk 3]
```diff
@@ -0,0 +1,139 @@
+import { Injectable } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import axios from 'axios';
+
+@Injectable()
+export class KeycloakService {
```
[the remark this answers is not in the document]
Comment (jaspalsingh-aot) [reply backend/users/src/app/services/KeycloakService.service.ts]: changed as suggested.

File: backend/users/src/app/controllers/users.controller.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
+import { Controller, Post, Body, HttpException, HttpStatus  } from '@nestjs/common';
+import { Unprotected } from 'nest-keycloak-connect';
+import { KeycloakService } from 'src/app/services/KeycloakService.service';
+import { AddUserToGroupDto } from 'src/app/dto/addUserToGroup';
+
+@Controller('users')
+@Unprotected()
```
[the remark this answers is not in the document]
Comment (jaspalsingh-aot) [reply backend/users/src/app/controllers/users.controller.ts]: working on it.

**Review by nikhila-aot — APPROVED** [review]

**Pull request merged**